### PR TITLE
BigQuery: deprecate `client.dataset()` part 4 (old)

### PR DESCRIPTION
**Pull request by Emar-Kar**
This PR contains five snippets ([full list](https://github.com/q-logic/google-cloud-python/pull/27#issue-305969319)):
- client_query_destination_table_cmek
- client_query_destination_table_legacy
- client_query_dry_run
- client_query_legacy_sql
- client_query_relax_column

### CODE REVIEW

File: bigquery/samples/tests/test_client_query_legacy_sql.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+from .. import client_query_legacy_sql
+
+
+def test_client_query_legacy_sql(capsys, client):
+
+    client_query_legacy_sql.client_query_legacy_sql(client)
+    out, err = capsys.readouterr()
+    assert "The query data:" in out
```
Comment (IlyaFaer) [inline bigquery/samples/tests/test_client_query_legacy_sql.py]: It tests only title print. Why not to check rows output?

File: bigquery/docs/usage/queries.rst [hunk 2]
```diff
@@ -35,6 +44,32 @@ See BigQuery documentation for more information on
    :start-after: [START bigquery_query_destination_table]
    :end-before: [END bigquery_query_destination_table]
 
+Write a query results in the legacy SQL syntax to a destination table with the
+:func:`~google.cloud.bigquery.client.Client.query` method:
```
Comment (IlyaFaer) [inline bigquery/docs/usage/queries.rst]: I think, query itself written in legacy SQL syntax, and in that comment it sounds like you're talking about result. Result represents several rows, they don't have any syntax, only fields and values. So this comment should be rephrased

File: bigquery/samples/client_query_destination_table_legacy.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    query_job = client.query(
+        sql,
+        # Location must match that of the dataset(s) referenced in the query
+        # and of the destination table.
+        location="US",
+        job_config=job_config,
+    )  # API request - starts the query
+
+    query_job.result()  # Waits for the query to finish
+    if query_job.state == "DONE":
```
Comment (IlyaFaer) [inline bigquery/samples/client_query_destination_table_legacy.py]: What if server will be overloaded with jobs, and our job will be processed too slowly? Test will fail. I'd look for a way to wait for job to be fininshed. Seems to me, you already used such a method: `query_job.result()`

File: bigquery/samples/client_query_dry_run.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        location="US",
+        job_config=job_config,
+    )  # API request
+
+    # A dry run query completes immediately.
+    if (
+        query_job.state == "DONE"
+        and query_job.dry_run
+        and query_job.total_bytes_processed > 0
+    ):
```
Comment (IlyaFaer) [inline bigquery/samples/client_query_dry_run.py]: In here it's better to wait for job to finished too, or it'll be breaking down test in some occasions

**Comment by Emar-Kar** [comment]
reformatted

**Pull request closed without merging**